File: feature_tests/java/somelib/src/main/java/dev/diplomattest/somelib/ErrorEnum.java
package dev.diplomattest.somelib;

import dev.diplomattest.somelib.ntv.somelib_h;


import java.lang.foreign.Arena;
import java.lang.foreign.MemorySegment;
import java.lang.ref.Cleaner;
import java.lang.foreign.SegmentAllocator;
import static java.lang.foreign.ValueLayout.*;
import java.nio.charset.StandardCharsets;

public enum ErrorEnum {
    Foo,
    Bar,
    ;

    static ErrorEnum fromInt(int i) {
        switch (i) {
            case 0 -> {
                return ErrorEnum.Foo;
            }
            case 1 -> {
                return ErrorEnum.Bar;
            }
            
        }
        throw new RuntimeException("Unexpected int for ErrorEnum:" + i);
    }

    int toInt() {
        switch (this) {
            case Foo -> {
                return 0;
            }
            case Bar -> {
                return 1;
            }
            
        }
        throw new RuntimeException("Unexpected variant for ErrorEnum:" + this);
    }

    
}

File: feature_tests/kotlin/somelib/src/main/kotlin/dev/diplomattest/somelib/ErrorEnum.kt
package dev.diplomattest.somelib

import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface ErrorEnumLib: Library {
}
enum class ErrorEnum {
    Foo,
    Bar;

    fun toNative(): Int {
        return this.ordinal
    }


    companion object {
        internal val libClass: Class<ErrorEnumLib> = ErrorEnumLib::class.java
        internal val lib: ErrorEnumLib = Native.load("somelib", libClass) 
        fun fromNative(native: Int): ErrorEnum {
            return ErrorEnum.entries[native]
        }

        fun default(): ErrorEnum {
            return Foo
        }
    }
}
class ErrorEnumError internal constructor(internal val value: ErrorEnum): Exception("Rust error result for ErrorEnum") {
    override fun toString(): String {
        return "ErrorEnum error with value " + value
    }

    fun getValue(): ErrorEnum = value
}